packages/backend/src/controllers/projectController.ts

```diff
@@ -505,4 +513,109 @@ export class ProjectController extends BaseController {
             results: undefined,
         };
     }
+
+    @Middlewares([allowApiKeyAuthentication, isAuthenticated])
+    @SuccessResponse('200', 'Success')
+    @Get('{projectUuid}/dashboards')
+    @OperationId('getDashboards')
+    async getDashboards(
+        @Path() projectUuid: string,
+        @Request() req: express.Request,
+    ): Promise<ApiGetDashboardsResponse> {
+        this.setStatus(200);
+
+        const chartUuid: string | undefined =
+            typeof req.query.chartUuid === 'string'
+                ? req.query.chartUuid.toString()
+                : undefined;
+
+        const includePrivate = req.query.includePrivate !== 'false';
+
+        const results = await this.services
+            .getDashboardService()
+            .getAllByProject(req.user!, projectUuid, chartUuid, includePrivate);
+
+        return {
+            status: 'ok',
+            results,
+        };
+    }
+
+    @Middlewares([
+        allowApiKeyAuthentication,
+        isAuthenticated,
+        unauthorisedInDemo,
+    ])
+    @SuccessResponse('201', 'Created')
+    @Post('{projectUuid}/dashboards')
+    @OperationId('createDashboard')
+    async createDashboard(
+        @Path() projectUuid: string,
+        @Body() body: DuplicateDashboardParams | CreateDashboard,
+        @Request() req: express.Request,
+        @Query() duplicateFrom?: string,
+    ): Promise<ApiCreateDashboardResponse> {
+        const dashboardService = this.services.getDashboardService();
+        this.setStatus(201);
+
+        let results: ApiCreateDashboardResponse['results'];
+
+        if (duplicateFrom) {
+            if (!isDuplicateDashboardParams(body)) {
+                throw new ParameterError(
+                    'Invalid parameters to duplicate dashbaord',
+                );
+            }
+
+            results = await dashboardService.duplicate(
+                req.user!,
+                projectUuid,
+                duplicateFrom.toString(),
+                body,
+            );
+        } else {
+            if (isDuplicateDashboardParams(body)) {
+                throw new ParameterError(
+                    'Invalid parameters to duplicate dashbaord',
+                );
+            }
+
+            results = await dashboardService.create(
+                req.user!,
+                projectUuid,
+                body,
+            );
+        }
+
+        return {
+            status: 'ok',
+            results,
+        };
+    }
+
+    @Middlewares([
+        allowApiKeyAuthentication,
+        isAuthenticated,
+        unauthorisedInDemo,
+    ])
+    @SuccessResponse('200', 'Updated')
+    @Patch('{projectUuid}/dashboards')
+    @OperationId('updateDashboards')
+    async updateDashboards(
+        @Path() projectUuid: string,
+        @Body() body: UpdateMultipleDashboards[],
```

Comment:
Update multiple dashboards extends Dashboard, which contains DashboardFilters :fire: This type has caused issues in the past because of the Record (eg: /calculate-total simply uses any instead)

Reply:
IT seems this endpoint is only used when moving dashboards to a space 
![image](https://github.com/user-attachments/assets/c5436f57-e4b9-40a5-809f-540f6bd7fbac)